src/App.tsx

```diff
@@ -9,58 +9,87 @@ import 'hds-core';
 import querystring, { ParsedUrlQuery } from 'querystring';
 import {
   AuthTokens,
-  setTokens,
+  storeTokens,
   getTokens,
+  removeTokens,
   convertParamsToTokens,
   isValidAuthParams,
   AuthContext,
 } from './auth/auth-context';
+import api from './common/utils/api/api';
 import Main from './components/main/Main';
 import NavigationAndFooterWrapper from './components/navigation-and-footer-wrapper/NavigationAndFooterWrapper';
 import './App.scss';
 import TargetPage from './target/page/TargetPage';
 
+type OptionalAuthTokens = AuthTokens | undefined;
+
 export default function App(): JSX.Element {
-  const existingAuthTokens: AuthTokens | undefined = getTokens();
+  const [isLoading, setLoading] = useState<boolean>(true);
+  const [authTokens, setAuthTokens] = useState<OptionalAuthTokens>();
 
-  const queryParams: ParsedUrlQuery = querystring.parse(
-    window.location.search.replace('?', '')
-  );
+  useEffect(() => {
+    const storedAuthTokens: OptionalAuthTokens = getTokens();
+    const queryParams: ParsedUrlQuery = querystring.parse(
+      window.location.search.replace('?', '')
+    );
 
-  const authTokensFromQuery = isValidAuthParams(queryParams)
-    ? convertParamsToTokens(queryParams)
-    : undefined;
+    const authTokensFromQuery: OptionalAuthTokens = isValidAuthParams(
+      queryParams
+    )
+      ? convertParamsToTokens(queryParams)
+      : undefined;
 
-  const authTokens: AuthTokens | undefined =
-    authTokensFromQuery || existingAuthTokens;
+    const authTokensFromQueryOrStore: OptionalAuthTokens =
+      authTokensFromQuery || storedAuthTokens;
 
-  useEffect(() => {
-    if (authTokens) {
-      setTokens(authTokens);
+    if (authTokensFromQueryOrStore) {
+      api
+        .testAuthCredentials(authTokensFromQueryOrStore)
```

Comment:
Why do we need to separately test these tokens on page load? Is testing not done while we retrieve the info of the target(resource)?

Reply:
Discussed about this on slack. We need to make a separate auth check because most of the GET apis are public, and accessible even without any auth.